### DISCUSSION TITLE
Welcome to iLogic-Vault Discussions!

### DISCUSSION THREAD

**Discussion by koechlm** [koechlm/iLogic-Vault#10] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by markalmack** [comment]
Hi Marcus,

I've downloaded your Source Code (iLogic-Vault-2020.4), but can't find the "QuickstartiLogicLibrary.dll" file in there. 

I can see "iLogic-Vault QuickstartLibrary.cs", "iLogic-Vault QuickstartLibrary.csproj" and "iLogic-Vault QuickstartLibrary.user". 

Am I missing something?

regards

Mark

**Reply by koechlm** [reply]
Hi Mark,
The source code is meant to allow compiling your own extension. I am sharing binaries and a setup on the releases page of this repository (https://github.com/koechlm/iLogic-Vault/releases/latest). However, - "Release" does not target an application to be installed in production environments; it is meant as an evaluation sample. Review the disclaimer for more details.
Thanks,
Markus

**Comment by DariuszMrozik** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.
> * Share ideas.
> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

Hi Marcus,
Is there a oportunity to check in file to Vault but for specyfic file (inventor document)? For now if I understand correctly user can only check in files by VaultCheckinTop comandManager control definition. Best will be if we can add in iLogic comment to check in and list of files / Inventor documents.

**Reply by koechlm** [reply]
Hi Dariusz,
the upcoming update includes AddFile for Non-CAD; this allows to upload of exported files, like BOM export or STEP to Vault. A direct upload would require to share Inventor-Vaul libraries internally to iLogic. I suggest posting it to the idea station.

**Comment by HartmutA** [comment]
Moin Markus,
i tried to add a function to get the parents from vault. The function works well, i'll share it later.
I have trouble with the version of the vault dll's after updating to 2022.2.
I removed the references and added them new from the vault client folder.
The version from the file is 27.2.67.0. After assigning the version in the properties is shown as 27.0.0.0
(hope you can see the the attached picture)
The problem is, i have share the right version.
Do you have an idea where i can correct this?
Greetings from Hartmut, MuM Hamburg.
![2021-12-17 15_06_52-Window](https://user-images.githubusercontent.com/62034753/146558682-73546d89-b701-4d6d-a4da-ee4c0e3ad2a9.png)

**Reply by koechlm** [reply]
Hello Hartmut @HartmutA: I run into a similar experience, whereas I had referenced update 1 libraries. I resolved the issue referencing the 2022 RTM SDK Libraries. The version of the libraries is expected to be 27.0.0.0 as the updates did not change APIs.
Thanks,
Markus

**Comment by ConstantVariable6105** [comment]
Markus,

Is this utility compatible with Inventor Professional 2022.4.1 (Build 492) and Vault **Basic** 2022.4.1 (Build 27.4.13.0)?

I downloaded the Quickstart source code for 2022.
I updated the reference paths for _Autodesk.Connectivity.Explorer.ExtensibilityTools_ and _Autodesk.Connectivity.WebServices_.
I built the solution successfully.
I copied the DLL and XML files to the iLogicAdd folder.
Followed the other instructions for iLogic paths, snippets, rules, etc.
I restarted Inventor.

I wrote a small iLogic program. I included the AddReference to the DLL in the iLogic code. 

However, when I try to run a simple iLogic code, Inventor reports:

 'iLogicVault' is not declared.

Thank you for your time and attention. I look forward to your reply.

Regards,
Jerry

**Reply by koechlm** [reply]
Hi Jerry, You need to update the Inventor reference as well. You also need to run Vault Professional; iLogic-Vault is not compatible with Vault Basic.
Thanks
Markus

**Comment by ConstantVariable6105** [comment]
Acknowledged. Thank you, Markus.

Regards,
Jerry

**Comment by HEGVaultjobserver** [comment]
Hello Markus,

I would be interested in installing the 2025 version.
Is there any estimate date for its release?

Best Regards,
Francisco Vaz

**Reply by koechlm** [reply]
Hi Francisco,
As stated in the Readme, iLogic-Vault moved into the Inventor core product starting in Inventor 2024. Therefore, I don't plan future releases here.
Best regards,
Markus
